nodejs/aws-infra/iam/index.ts

```diff
@@ -0,0 +1,195 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+import * as utils from "./../utils";
+
+
+export class User extends pulumi.ComponentResource {
+    public readonly user: aws.iam.User;
+    public readonly accessKeys: aws.iam.AccessKey[];
+    public readonly groupMemberships: aws.iam.UserGroupMembership[];
+
+    constructor(name: string, args: UserArgs = {}, opts?: pulumi.CustomResourceOptions) {
+        super(`awsinfra:x:iam:User`, name, args, opts);
+
+        // Save group membership and access key args so we can define them inline.
+        const groupMembership = args.groupMembership;
+
+        // Explicitly delete these props so we do *not* pass them into the User created
+        // below.
+        delete args.groupMembership;
+
+        this.user = args.user || new aws.iam.User(name, args, { parent: this });
+
+        this.accessKeys = [];
+
+        this.groupMemberships = [];
+        if (groupMembership !== undefined) {
+            this.groupMemberships.push(
+                new aws.iam.UserGroupMembership(
+                    name,
+                    { ...groupMembership, user: this.user.name },
+                    { parent: this },
+                ),
+            );
+        }
+    }
+
+    /**
+     * Create an AccessKey for the current `User`.
+     *
+     * @param name The _unique_ name of the resulting `AccessKey` resource.
+     * @param args The properties of the resulting `AccessKey` resource.
+     * @param opts A bag of options that control the resulting `AccessKey` resource's behavior.
+     * __NOTE__: the `parent` field is overridden and set to `this`, rather than the value (if any)
+     * in `opts`.
+     */
+    public createAccessKey(
+        name: string,
+        args?: AccessKeyArgs,
+        opts?: pulumi.CustomResourceOptions,
+    ): aws.iam.AccessKey {
+        const key = new aws.iam.AccessKey(
+            name,
+            { ...args, user: this.user.name },
+            { ...opts, parent: this },
```

Comment:
our pattern elsewhere has been to do:  `opts || { parent: this }`   But yours is arguably more correct.  most correct would likely be: `{ parent: this, ...opts}`.  i.e. we try to fill in useful default values, but the caller always has the final say.

That said, can you use `opts || { parent: this }` for now (to be consistent with the rest of aws-infra).  I'll make a change to update these all to be better.